examples/ilu-preconditioned-solver/ilu-preconditioned-solver.cpp

```diff
@@ -92,10 +97,10 @@ int main(int argc, char *argv[])
     // if there are several very similar systems to solve, and the same
     // solver+preconditioner combination is expected to be effective.
     auto ilu_gmres_factory =
-        gko::solver::Gmres<>::build()
+        gmres::build()
             .with_criteria(
                 gko::stop::Iteration::build().with_max_iters(1000u).on(exec),
-                gko::stop::ResidualNormReduction<>::build()
+                gko::stop::ResidualNormReduction<ValueType>::build()
                     .with_reduction_factor(1e-15)
```

Comment:
This probably needs to be adaptated. You can use something like the r<valuetype>::value from the tests.

Reply:
This is the same basically everywhere, not just here.

Reply:
Is it okay if I use a `ValueType(1e-15)` ? I dont want to complicate it with the `r::value` declarations. I guess to be uniform, I can use `1e-7` for everything as that makes sense for both `double` and `float`.

Reply:
You can use the C++ [`std::numeric_limits<ValueType>::digits10`](https://en.cppreference.com/w/cpp/types/numeric_limits/digits10), which should do basically the same that `r::value` does, just that it is from the standard library.

Reply:
Sorry, `digits10` is probably not the best here, [`std::numeric_limit<ValueType>::epsilon`](https://en.cppreference.com/w/cpp/types/numeric_limits/epsilon) is a lot more fitting.